src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -796,6 +817,215 @@ void Transformations::MainSnippets(void) {
+#if defined(OPENVINO_ARCH_ARM64)
+            return false;
+#else
+            // Broadcast is supported only for MHA tokenization where there are needed and special checks
+            if (auto broadcast_v1 = ov::as_type_ptr<const ov::op::v1::Broadcast>(n)) {
+                return broadcast_v1->get_broadcast_spec().m_type == ov::op::AutoBroadcastType::NUMPY;
+            } else if (auto broadcast_v3 = ov::as_type_ptr<const ov::op::v3::Broadcast>(n)) {
+                return broadcast_v3->get_broadcast_spec().m_type == ov::op::BroadcastType::NUMPY;
+            }
+            return false;
+#endif
+        };
+
+        auto is_supported_reduce_op = [](const std::shared_ptr<const ov::Node> &n) -> bool {
+#if defined(OPENVINO_ARCH_ARM64)
+            return false;
+#else
+            if (ov::is_type<const ov::op::v1::ReduceMax>(n) || ov::is_type<const ov::op::v1::ReduceSum>(n)) {
+                const auto& reduce_base = ov::as_type_ptr<const ov::op::util::ArithmeticReductionKeepDims>(n);
+                const auto& axis_constant = ov::as_type_ptr<const ov::op::v0::Constant>(n->get_input_node_shared_ptr(1));
+                const auto rank = n->get_input_partial_shape(0).rank();
+                if (rank.is_dynamic() || !reduce_base->get_keep_dims() || !axis_constant || shape_size(axis_constant->get_shape()) != 1)
+                    return false;
+
+                const auto axis_value = axis_constant->cast_vector<int32_t>(1)[0];
+                const auto normalized_axis = ov::util::normalize_axis(n->get_friendly_name(), axis_value, rank);
+                // Note: Reduction only over the last dimension is currently supported
+                return normalized_axis == rank.get_length() - 1;
+            }
+            return false;
+#endif
+        };
+
+        return is_supported_fq_op(n) ||
+            is_supported_unary_eltwise_op(n) ||
+            is_supported_binary_eltwise_op(n) ||
+            is_supported_ternary_eltwise_op(n) ||
+            is_supported_transpose(n) ||
+            is_supported_softmax(n) ||
+            is_supported_matmul(n) ||
+            is_supported_broadcast_op(n) ||
+            is_supported_reduce_op(n);
+    };
+
+    auto has_supported_in_out = [](const std::shared_ptr<const ov::Node> &n) -> bool {
+        auto supported = [&n](descriptor::Tensor& t) -> bool {
+            auto get_supported_element_types = []() {
+#if defined(OPENVINO_ARCH_ARM64)
+                static const std::set<ov::element::Type> supported_element_types =
+                    { ov::element::f32 };
+#else
+                static const std::set<ov::element::Type> supported_element_types =
+                    { ov::element::f32, ov::element::bf16, ov::element::i8, ov::element::u8 };
+#endif
+                return supported_element_types;
+            };
+            // TODO [122585] Need to add dynamic rank support
+            if (t.get_partial_shape().rank().is_dynamic())
```

Comment:
As I wrote in the comment in `src/common/snippets/src/pass/collapse_subgraph.cpp` - it's a common check which must be in Snippets tokenization pass. Here you should check only precisions.

Reply:
Applied. Thanks Alexandra!